forge-game/src/main/java/forge/game/ability/effects/ProtectAllEffect.java

```diff
@@ -87,6 +88,7 @@ public void resolve(SpellAbility sa) {
 
             for (final Card tgtC : list) {
                 tgtC.addChangedCardKeywords(gainsKWList, null, false, timestamp, null, true);
+                game.fireEvent(new GameEventCardStatsChanged(tgtC));
```

Comment:
Why not do this event in the addChangedCardKeywords function?

Reply:
because most other effects like Pump can do more than just add KW, so while the View properties should be updated it's better to only call the event once at the end

Reply:
I still think the event should be part of a function

Reply:
Hmn, not sure it helps to override it everywhere when usually the effect will have to do the work instead e.g.:

![image](https://github.com/user-attachments/assets/ccc151a7-cdbb-4158-b385-cfbc54835120)

I might do more work on the events later, maybe then it becomes easier 🤔